code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -1150,7 +1150,7 @@ func (sab *storageAllocationBase) cancellationCharge(cancellationFraction float6
 	return currency.MultFloat64(cost, cancellationFraction)
 }
 
-func (sa *storageAllocationBase) requiredTokensForUpdateAllocation(cpBalance currency.Coin, extend, isEnterprise bool, now common.Timestamp) (currency.Coin, error) {
+func (sa *storageAllocationBase) requiredTokensForUpdateAllocation(herculesEnabled bool, cpBalance currency.Coin, extend, isEnterprise bool, now common.Timestamp) (currency.Coin, error) {
```

Comment:
create 2 different functions (better copypaste) , it doesn't look good if we escape the fork logic